peps/pep-9999.rst

```diff
@@ -0,0 +1,616 @@
+so it only introduces compatibility concerns if tools choose to adopt an
+index behavior other than the behavior(s) they currently implement.
+
+This PEP’s language does not quite align with existing tools, including
+pip and uv. Either this PEP’s language can change during review, or if
+this PEP’s language is preferred, other projects could conform to it.
+The important thing is that all the projects are using the same language
+for the same concepts.
+
+As some tools rely on one or the other behavior, there are some possible
+issues that may emerge, where tailoring available resources/packages for
+a particular behavior may detract from the user experience for people
+who rely on the other behavior.
+
+- Different repos may have different metadata. For example, one cannot
+  assume that the metadata for package “something” on repository “A” has
+  the same dependencies as “something” on repository “B”. When an
+  installer falls back to a different repository in the search order, it
+  implies refreshing the package metadata from the new repository. This
+  is both an improvement and a complication. It is a complication in the
+  sense that a cached metadata entry must be keyed by both package name
+  and index url, instead of just package name. It is a potential
+  improvement in that different implementation variants of a package can
+  differ in dependencies as long as their distributions are separated
+  into different indexes.
+- Users may not get updates as they expect, because some higher priority
+  index has not updated/synchronized with PyPI to get the latest
+  packages. If the higher priority index has a valid candidate, newer
+  packages will not be found. This will need to be communicated
+  verbosely, because it is counter to pip’s well-established behavior.
+- Improving the predictability of which repository will be selected may
+  tempt people into using index priority as a way of having similarly
+  named files that have different contents. It would be helpful if tools
+  errored on mismatching hashes, or otherwise gave the user feedback
+  that they are breaking a key assumption. This PEP does not mandate
+  alternative fixes, such as augmenting the cache key with the index,
+  because these changes have unknown unintended consequences. However,
+  it is advisable to augment the cache key with the index, as doing so
+  would remedy known “gotchas.”
+
+Security Implications
+=====================
+
+Index priority creates a mechanism for users to explicitly specify a
+trust hierarchy among their indexes. As such, it limits the potential
+for dependency confusion attacks. Index priority was `rejected by PEP
+708 <https://peps.python.org/pep-0708/#provide-a-mechanism-to-order-the-repositories>`__
+as a solution for dependency confusion attacks. This PEP requests that
+the rejection be reconsidered, with index priority serving a different
+purpose. This PEP is primarily motivated by the desire to support
```

Comment:
See, this PEP _does_ kind of promote one of its anti-goals 😄 - which I think is a good thing!